src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml

```diff
@@ -1,10 +1,30 @@
+parameters:
+    amsterdamphp_site.integration.youtube_factory.class: AmsterdamPHP\Bundle\SiteBundle\Service\YoutubeFactory
+    amsterdamphp_site.integration.youtube_client.class: Madcoda\Youtube
+
 services:
+    amsterdamphp_site.integration.youtube_factory:
+        class: %amsterdamphp_site.integration.youtube_factory.class%
+        arguments:
+            key: %youtube_key%
+
+    amsterdamphp_site.integration.youtube_client:
+        class:            "%amsterdamphp_site.integration.youtube_client.class%"
+        factory_service:  amsterdamphp_site.integration.youtube_factory
+        factory_method:   getClient
```

Comment:
Ok i'm fine with this, even if you could drop the factory and just intantiate the client directly as a service, but this gives us a place to set defaults if we ever need them.